### PR TITLE
metal: build ios shaders in Makefile

### LINKED ISSUE #3086: Missing iOS metallib
Short info header:
- GFX version: 0.4
- OS: iOS 13.13
- GPU: iphone 7 gpu

```rust
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: "This library format is not supported on this platform (or was built with an old version of the tools)"', src/libcore/result.rs:1165:5
stack backtrace:
   0:        0x10504c004 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h406e22a529a915ed
   1:        0x10507c618 - core::fmt::write::h4f487e714088986d
   2:        0x105040888 - std::io::Write::write_fmt::hea82d7c79e61f422
   3:        0x10504f79c - std::panicking::default_hook::{{closure}}::haa6f68d0d27fe3d8
   4:        0x10504f498 - std::panicking::default_hook::h2be19e15ac4eb384
   5:        0x10504fda0 - std::panicking::rust_panic_with_hook::haac78557dcef3a9f
   6:        0x10504fa20 - std::panicking::continue_panic_fmt::h588e812853cb4f12
   7:        0x10504f988 - rust_begin_unwind
   8:        0x10509ac80 - core::panicking::panic_fmt::h09741a3213dba543
   9:        0x10509ad8c - core::result::unwrap_failed::hf47c31c429b02014
  10:        0x104aef0dc - core::result::Result<T,E>::unwrap::h1d853f22ffbbb8ee
  11:        0x104b2aaa8 - gfx_backend_metal::internal::ServicePipes::new::hb04c8b4e4ee22f99
  12:        0x104ad814c - gfx_backend_metal::Shared::new::hf3d2a3e0aaf00964
  13:        0x104ad8650 - <gfx_backend_metal::Instance as gfx_hal::Instance<gfx_backend_metal::Backend>>::enumerate_adapters::{{closure}}::heaa93e7bcfb09db8
  14:        0x104ab70ac - core::iter::adapters::map_fold::{{closure}}::h2ffa82e44adbd2ab
  15:        0x104a7f670 - core::iter::traits::iterator::Iterator::fold::ok::{{closure}}::h7daea70248fcddc9
  16:        0x104afcf18 - core::iter::traits::iterator::Iterator::try_fold::h9e1b67d0e4266096
  17:        0x104afccb4 - core::iter::traits::iterator::Iterator::fold::he195bcb76df91f36
  18:        0x104abbbdc - <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold::h2f335fa5220d6201
  19:        0x104ab4cc0 - core::iter::traits::iterator::Iterator::for_each::h47a2bf9f549444f6
  20:        0x104b0c718 - <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend::h147f9ae965b49662
  21:        0x104b0f8c8 - <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter::hff5d2f6b6942605e
  22:        0x104b133c0 - <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter::h31359f896d46316b
  23:        0x104ab493c - core::iter::traits::iterator::Iterator::collect::h20b311a90b090893
  24:        0x104ad8550 - <gfx_backend_metal::Instance as gfx_hal::Instance<gfx_backend_metal::Backend>>::enumerate_adapters::h0fe3442a7892529a
  25:        0x1049e7280 - wgpu_native::instance::request_adapter::h544681132a2c3ab4
  26:        0x1049e7e2c - wgpu_request_adapter
  27:        0x1047613b8 - wgpu::Adapter::request::h78fcdc88660ffd94
  28:        0x10437f49c - game_lib::framework::run::h4297965554738eb9
  29:        0x1043826bc - game_lib::main_rs::ha566112f155e7bc4
  30:        0x10437539c - core::ops::function::FnOnce::call_once::h8d12fb123e67e7fc
  31:        0x104375860 - std::panicking::try::do_call::h40f2bebd843e3f3e
  32:        0x10505c754 - __rust_maybe_catch_panic
  33:        0x104375710 - std::panicking::try::haee8726b161251bc
  34:        0x1043756cc - std::panic::catch_unwind::h7dfda187ea2f0e6a
  35:        0x1043751f0 - game_bindings::stop_unwind::h0449398b653afb0c
  36:        0x104375470 - main_rs
  37:        0x10437503c - main
Attempt to Unwind out of rust code

```

### PR DESCRIPTION
**Pull request by Gordon-F**
Fixes #3086
PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [ ] tested examples with the following backends:
- [ ] `rustfmt` run on changed code

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,7 +39,7 @@ else
 endif
 
 
-.PHONY: all check quad quad-wasm test doc reftests benches shader-binaries
+.PHONY: all check quad quad-wasm test doc reftests benches shader-binaries shader-binaries-ios
```
Comment (kvark) [inline Makefile]: why don't we build both macos and ios in the `shader-binaries` command?

File: src/backend/metal/src/internal.rs [hunk 2]
```diff
@@ -451,7 +451,10 @@ pub struct ServicePipes {
 
 impl ServicePipes {
     pub fn new(device: &metal::DeviceRef) -> Self {
+        #[cfg(target_os = "macos")]
         let data = include_bytes!("./../shaders/gfx_shaders.metallib");
```
Comment (kvark) [inline src/backend/metal/src/internal.rs]: nit: we could also do `if cfg!(target_os = "macos") { "../shaders/gfx_shaders.metallib" } else { "../shaders/gfx_shaders_ios.metallib" }` as an expression

Comment (kvark) [inline src/backend/metal/src/internal.rs]: let's rename the old one for consistency to `gfx_shaders-macos.metallib`.
Perhaps, we could use the dashes in both even? i.e. `gfx-shader-macos.metallib`

Comment (Gordon-F) [reply src/backend/metal/src/internal.rs]: @kvark how i can implement this?

If i do this:
```rust
let data = if cfg!(target_os = "macos") {
            include_bytes!("./../shaders/gfx-shaders-macos.metallib")
        } else {
            include_bytes!("./../shaders/gfx-shaders-ios.metallib")
        };
```
I get compiler error - `if and else have incompatible types`

File: Makefile [hunk 3]
```diff
@@ -80,8 +80,15 @@ quad-wasm:
 
 shader-binaries:
 ifeq ($(UNAME_S),Darwin)
+	# MacOS
 	cd ./src/backend/metal/shaders && \
 	xcrun -sdk macosx metal -c *.metal -mmacosx-version-min=10.11 && \
-	xcrun -sdk macosx metallib *.air -o gfx_shaders.metallib && \
+	xcrun -sdk macosx metallib *.air -o gfx-shaders-macos.metallib && \
+	rm *.air
+
```
Comment (kvark) [inline Makefile]: let's not put empty lines inside the makefile commands please

File: src/backend/metal/src/internal.rs (hunk 2 above)
Comment (kvark) [reply src/backend/metal/src/internal.rs]: I guess it would have to be `&include_bytes!(...)[..]`

**Comment by Gordon-F** [comment]
@kvark all done. Can i also make PR in 0.4 branch ?

**Comment by kvark** [comment]
> Can i also make PR in 0.4 branch ?

That would be great, yes!

**Pull request merged**